src/background.js

```diff
@@ -4,9 +4,66 @@ import 'src/activity-logger/background'
 import 'src/omnibar'
 import { installTimeStorageKey } from 'src/imports/background'
 import convertOldData from 'src/util/old-data-converter'
+import fetchPageData from 'src/page-analysis/background/fetch-page-data'
+import { findPagesByUrl } from 'src/search/find-pages'
+import { transformToBookmarkDoc } from 'src/imports/background/imports-preparation'
+import { dataURLToBlob } from 'blob-util'
+import { generatePageDocId } from 'src/page-storage'
+import db from 'src/pouchdb'
 
 export const dataConvertTimeKey = 'data-conversion-timestamp'
 
+
+export async function bookmarkStorageListener(id, bookmarkInfo) {
+    const urlToFind = bookmarkInfo.url
+    const samePageCandidates = (await findPagesByUrl({urlToFind})).rows.map(row => row.doc)
+    if (samePageCandidates.length > 0) {
+        let pageDoc = {}
+        for (var i = samePageCandidates.length - 1; i >= 0; i--) {
+            if (samePageCandidates[i].url === bookmarkInfo.url) {
+                pageDoc = samePageCandidates[i]
+                break
+            }
+        }
+        const bookmarkDoc = await transformToBookmarkDoc(pageDoc)(bookmarkInfo)
+        db.bulkDocs([bookmarkDoc, pageDoc])
+        return
+    }
+    let pageDoc = {
+        url: bookmarkInfo.url,
+        title: bookmarkInfo.title,
+        _id: generatePageDocId(bookmarkInfo.dateAdded, id),
+    }
+    try {
+        console.log("Trying to get page data")
+        const freezeDryFlag = window.localStorage.getItem('freeze-dry-bookmarks')
+        let fetchPageDataOptions = {
+            includePageContent: true,
+            includeFavIcon: true,
+        }
+        if (freezeDryFlag) {
+            fetchPageDataOptions.includeFreezeDry = true
+        }
+        const pageData = await fetchPageData(bookmarkInfo.url, fetchPageDataOptions)
```

Comment:
Have a look at the function signature for `fetchPageData`. It accepts an object of arguments, so you would call it something like `fetchPageData({ url: bookmarkInfo.url, opts: fetchPageDataOptions })`